apps/dashboard/lib/trpc/routers/key/updateRemaining.ts

```diff
@@ -48,14 +43,11 @@ export const updateKeyRemaining = t.procedure
         await tx
           .update(schema.keys)
           .set({
+            refillInterval: input.refill?.refillDay ? "monthly" : "daily",
             remaining: input.remaining ?? null,
-            refillInterval:
-              input.refill?.interval === "none" || input.refill?.interval === undefined
-                ? null
-                : input.refill?.interval,
-            refillDay: isMonthlyInterval ? input.refill?.refillDay : null,
+            refillDay: input.refill?.refillDay ?? null,
             refillAmount: input.refill?.amount ?? null,
-            lastRefillAt: input.refill?.interval ? new Date() : null,
+            lastRefillAt: input.refill?.amount ? new Date() : null,
```

Comment:
I'm not sure if that's possible, but same scenario applies here as well. If amount becomes `0` this will yield null.